#==================================================================================================
#
# Name: lldb - build fails with SWIG 4.1.0
# Date: 2022-09-21
# Link: https://bugzilla.redhat.com/show_bug.cgi?id=2128646
# 
# Summary:
#   Fix the failure caused by change in SwigValueWraper for C++11 and later
#   for improved move semantics in SWIG commit.
#
# Trac Ticket: https://trac.macports.org/ticket/67828
#
#==================================================================================================
--- a/tools/lldb/bindings/python/python-typemaps.swig
+++ b/tools/lldb/bindings/python/python-typemaps.swig
@@ -435,7 +435,7 @@ template <> bool SetNumberFromPyObject<double>(double &number, PyObject *obj) {
 
 %typemap(out) lldb::FileSP {
   $result = nullptr;
-  lldb::FileSP &sp = $1;
+  const lldb::FileSP &sp = $1;
   if (sp) {
     PythonFile pyfile = unwrapOrSetPythonException(PythonFile::FromFile(*sp));
     if (!pyfile.IsValid())
